gadgetapp/README.md

```diff
@@ -0,0 +1,118 @@
+# Gadget App
+
+The Gadget App is a browser based HMI that allows for nearly real time monitoring and control of the system. It's built using Django and a base level understanding of the technology and model-view-control design paradigm is useful.
+
+The Gadget App is connected to the rest of the system through the postgres database. When an inspection event occurs, the services in the inspection pipeline insert a row into the inspection events table. The Gadget App is consistently polling that table for a new event. When it sees a new event, it take the information the from the model pipeline and uses that to update the display. It is also monitoring the state of the other services and will update the display if that changes.
+
+Out of the box, the Gadget App comes configured for a single sensor, single pipeline system. It has one canvas and two charts. To connect it to a Gadget some changes will have to be made to these files:
+
+- models.py
+- views.py
+- forms.py
+- init_db.py
+
+Most of the the changes will be done in models.py. As mentioned previously, the Gadget App polls the inspection events table to for new events. To do that it need to know what service, or services, inspection events it needs to look for. That is specified using the **INSPECTION_RESULT_KEYS**, which can be found at the top of models.py. By default it looks like this:
+
+    INSPECTION_RESULT_KEYS={   
+        0:{'service_type':'pipeline','instance_name':'gadget-pipeline','instance':0,'sensor_topic':'sensor/gadget-sensor-gocator/0'}
+    }
+
+With this the Gadget App knows that it's looking for inspection events from the model pipeline service *gadget-pipeline* instance 0 that came from the sensor topic *sensor/gadget-sensor-gocator/0*. If any of those things have changed it will need to be updated here.
```

Comment:
``If any of those things have changed it will need to be updated here.``
need a comma